cms/static/js/factories/login.js

```diff
@@ -22,15 +22,19 @@ define(['jquery.cookie', 'utility'], function() {
 
         $('form#login_form').submit(function(event) {
             event.preventDefault();
+            var submit_button = $('#submit'),
+                deferred = new $.Deferred(),
+                promise = deferred.promise();
+            ViewUtils.disableElementWhileRunning(submit_button, function() { return promise; });
```

Comment:
@Shrhawk , I agree this code is going to be better understood if you follow andy's advice here :)